digicampipe/calib/camera/dl2.py

```diff
@@ -2,6 +2,26 @@
 import numpy as np
 import astropy.units as u
 
+def hillas_parameters(geom, image):
+    unit = geom.pix_x.unit
+    try:
+       moments = hillas.hillas_parameters(geom, image)
+    except hillas.HillasParameterizationError:
+        return hillas.MomentParameters(
+            size=None,
```

Comment:
Sorry but why None and not np.nan ?

Reply:
I don't know

Reply:
I do not know what is more apropriate here.

Reply:
I'd stick to None, as it suggests that it was not initialized. np.nan is ambiguous between "not initialized" and "wrongly computed" IMO

Reply:
It might be that the size is zero and for this reason all the other parameters cannot be calculated be calculated. But I am guess here, one would have to study the `hillas_parameter` function in detail and find the reason for the exception.

Reply:
Yes that's why it fails! 
https://github.com/cta-observatory/ctapipe/blob/b7a11b8c06c8f0140338974c262f58e650e6fcb7/ctapipe/image/hillas.py#L547-L548

Reply:
so should we just skip the event instead of returning a nan/None Container ?

Reply:
If we think of these generators as boxes in an hypothetical analysis process. And one of these boxes skips events, then all following boxes never see these events ... this includes boxes doing statistics ... boxes which might be an event display ... i don't know what. 

This might be pretty surprising for people.

I mean a box called "filter_this_and_that" ... clearly skips events. but a box called "calibrate _to_dl2" .. I'm not sure.

I guess this is a design question. One might say: All boxes just **add** information to the event. 
Some kind of "sink" in the end writes the results to disk, whatever was defined to be a result.

But we can also say: Boxes do whatever they like with the event_stream and people just have to read the docu, or the implementation of each box in case there is no docu.

Reply:
I think however, size should be zero, because this is really the case here.